### ISSUE TITLE
RegexpError: invalid backref number/name

### ISSUE THREAD

**Issue by chris-orchard** [dhoer/chef-windows_screenresolution#4]
I'm provisioning a Windows Server 2012 VM as a Selenium node using the selnium_grid cookbook, with vagrant and vagrant-berkshelf. After chef finishes Selenium is running but chef exits with the following error:

```
... truncated ...
13:40:49 ==> default: [2016-02-03T13:40:42+00:00] INFO: remote_directory[C:/var/chef/handlers] created directory C:/var/chef/handlers
13:40:49 ==> default: [2016-02-03T13:40:42+00:00] INFO: cookbook_file[C:/var/chef/handlers/windows_reboot_handler.rb] created file C:/var/chef/handlers/windows_reboot_handler.rb
13:40:49 ==> default: 
13:40:49 ==> default: [2016-02-03T13:40:42+00:00] INFO: cookbook_file[C:/var/chef/handlers/windows_reboot_handler.rb] updated file contents C:/var/chef/handlers/windows_reboot_handler.rb
13:40:49 ==> default: [2016-02-03T13:40:42+00:00] INFO: Disabling WindowsRebootHandler as a report handler.
13:40:49 ==> default: [2016-02-03T13:40:42+00:00] INFO: Enabling WindowsRebootHandler as a report handler.
13:40:49 ==> default: [2016-02-03T13:40:43+00:00] WARN: The windows_reboot resource is deprecated. Please use the reboot resource in
13:40:49 ==> default: 
13:40:49 ==> default: Chef Client 12. windows_reboot will be removed in the next major version
13:40:49 ==> default: release of the Windows cookbook.
13:40:49 ==> default: [2016-02-03T13:40:43+00:00] INFO: template[C:/selenium/config/selenium_node.json] created file C:/selenium/config/selenium_node.json
13:40:49 ==> default: 
13:40:49 ==> default: [2016-02-03T13:40:43+00:00] INFO: template[C:/selenium/config/selenium_node.json] updated file contents C:/selenium/config/selenium_node.json
13:40:49 ==> default: [2016-02-03T13:40:43+00:00] INFO: file[C:/selenium/bin/selenium_node.cmd] created file C:/selenium/bin/selenium_node.cmd
13:40:49 ==> default: [2016-02-03T13:40:43+00:00] INFO: file[C:/selenium/bin/selenium_node.cmd] updated file contents C:/selenium/bin/selenium_node.cmd
13:40:49 ==> default: [2016-02-03T13:40:43+00:00] INFO: file[C:/selenium/bin/selenium_node.cmd] not queuing delayed action request on windows_reboot[Reboot to start selenium_node] (delayed), as it's already been queued
13:40:49 ==> default: [2016-02-03T13:40:43+00:00] INFO: Added windows_shortcut[C:\Users\vagrant\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\Startup\selenium_node.lnk] shortcut
13:40:51 ==> default: [2016-02-03T13:40:45+00:00] INFO: execute[Firewall rule selenium_node for port 5555] ran successfully
13:40:51 ==> default: 
13:40:51 ==> default: [2016-02-03T13:40:45+00:00] INFO: template[C:/selenium/config/selenium_node.json] sending request action to windows_reboot[Reboot to start selenium_node] (delayed)
13:40:51 ==> default: [2016-02-03T13:40:46+00:00] INFO: user[rdp_local] created
13:40:51 ==> default: 
13:40:51 ==> default: [2016-02-03T13:40:46+00:00] INFO: Running queued delayed notifications before re-raising exception
13:40:51 ==> default: [2016-02-03T13:40:46+00:00] ERROR: Running exception handlers
13:40:51 ==> default: [2016-02-03T13:40:46+00:00] ERROR: Exception handlers complete
13:40:51 ==> default: [2016-02-03T13:40:46+00:00] FATAL: Stacktrace dumped to C:/var/chef/cache/chef-stacktrace.out
13:40:51 ==> default: [2016-02-03T13:40:46+00:00] FATAL: RegexpError: invalid backref number/name: /^C:\vagrant-berkshelf\5aebfec871579ca2a83dbf1303d0135a\cookbooks\windows_screenresolution\recipes\default.rb:([\d]+)/
13:40:51 Chef never successfully completed! Any errors should be visible in the
13:40:51 output above. Please fix your recipes so that they properly complete.
13:40:51 **** ERROR ON VAGRANT PROVISION ****
```

The stacktrace in C:/var/chef/cache/chef-stacktrace.out is:

```
RegexpError: invalid backref number/name: /^C:\vagrant-berkshelf\5aebfec871579ca2a83dbf1303d0135a\cookbooks\windows_screenresolution\recipes\default.rb:([\d]+)/
C:/opscode/chef/embedded/apps/chef/lib/chef/formatters/error_inspectors/resource_failure_inspector.rb:66:in `recipe_snippet'
C:/opscode/chef/embedded/apps/chef/lib/chef/formatters/error_inspectors/resource_failure_inspector.rb:43:in `add_explanation'
C:/opscode/chef/embedded/apps/chef/lib/chef/formatters/error_mapper.rb:72:in `resource_failed'
C:/opscode/chef/embedded/apps/chef/lib/chef/formatters/base.rb:139:in `resource_failed'
C:/opscode/chef/embedded/apps/chef/lib/chef/event_dispatch/dispatcher.rb:29:in `block (2 levels) in def_forwarding_method'
C:/opscode/chef/embedded/apps/chef/lib/chef/event_dispatch/dispatcher.rb:29:in `each'
C:/opscode/chef/embedded/apps/chef/lib/chef/event_dispatch/dispatcher.rb:29:in `block in def_forwarding_method'
C:/opscode/chef/embedded/apps/chef/lib/chef/resource.rb:574:in `rescue in run_action'
C:/opscode/chef/embedded/apps/chef/lib/chef/resource.rb:583:in `run_action'
C:/opscode/chef/embedded/apps/chef/lib/chef/runner.rb:49:in `run_action'
C:/opscode/chef/embedded/apps/chef/lib/chef/runner.rb:81:in `block (2 levels) in converge'
C:/opscode/chef/embedded/apps/chef/lib/chef/runner.rb:81:in `each'
C:/opscode/chef/embedded/apps/chef/lib/chef/runner.rb:81:in `block in converge'
C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/resource_list.rb:83:in `block in execute_each_resource'
C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:116:in `call'
C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:116:in `call_iterator_block'
C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:85:in `step'
C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:104:in `iterate'
C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:55:in `each_with_index'
C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/resource_list.rb:81:in `execute_each_resource'
C:/opscode/chef/embedded/apps/chef/lib/chef/runner.rb:80:in `converge'
C:/opscode/chef/embedded/apps/chef/lib/chef/client.rb:339:in `block in converge'
C:/opscode/chef/embedded/apps/chef/lib/chef/client.rb:334:in `catch'
C:/opscode/chef/embedded/apps/chef/lib/chef/client.rb:334:in `converge'
C:/opscode/chef/embedded/apps/chef/lib/chef/client.rb:353:in `converge_and_save'
C:/opscode/chef/embedded/apps/chef/lib/chef/client.rb:457:in `run'
C:/opscode/chef/embedded/apps/chef/lib/chef/application.rb:253:in `run_with_graceful_exit_option'
C:/opscode/chef/embedded/apps/chef/lib/chef/application.rb:230:in `block in run_chef_client'
C:/opscode/chef/embedded/apps/chef/lib/chef/local_mode.rb:39:in `with_server_connectivity'
C:/opscode/chef/embedded/apps/chef/lib/chef/application.rb:213:in `run_chef_client'
C:/opscode/chef/embedded/apps/chef/lib/chef/application/solo.rb:280:in `block in interval_run_chef_client'
C:/opscode/chef/embedded/apps/chef/lib/chef/application/solo.rb:269:in `loop'
C:/opscode/chef/embedded/apps/chef/lib/chef/application/solo.rb:269:in `interval_run_chef_client'
C:/opscode/chef/embedded/apps/chef/lib/chef/application/solo.rb:247:in `run_application'
C:/opscode/chef/embedded/apps/chef/lib/chef/application.rb:60:in `run'
C:/opscode/chef/embedded/apps/chef/bin/chef-solo:25:in `<top (required)>'
C:/opscode/chef/bin/chef-solo:63:in `load'
C:/opscode/chef/bin/chef-solo:63:in `<main>'
```

**Comment by dhoer** [comment]
What Chef version are you running and can you run Chef with debug on and submit results?

**Comment by chris-orchard** [comment]
Chef-client version on the windows VM is 12.3.0

I'll run with debug...

**Comment by dhoer** [comment]
That seems to be an issue with Chef 12.3:
https://github.com/chef/chef/pull/3357

It should be fixed in a later release of Chef, but I would like to see if it can be fixed to work with Chef 12.3.

**Comment by chris-orchard** [comment]
I turned debug on but am now getting a different error:

```
10:22:54 ==> default: [2016-02-04T10:22:52+00:00] INFO: user[rdp_local] created
10:22:54 ==> default: [2016-02-04T10:22:52+00:00] INFO: Processing group[Administrators] action modify (windows_screenresolution::default line 19)
10:22:54 ==> default: [2016-02-04T10:22:52+00:00] DEBUG: providers for generic group resource enabled on node include: [Chef::Provider::Group::Windows]
10:22:54 ==> default: [2016-02-04T10:22:52+00:00] DEBUG: providers that refused resource group[Administrators] were: []
10:22:54 ==> default: [2016-02-04T10:22:52+00:00] DEBUG: providers that support resource group[Administrators] include: [Chef::Provider::Group::Windows]
10:22:54 ==> default: [2016-02-04T10:22:52+00:00] DEBUG: providers that survived replacement include: [Chef::Provider::Group::Windows]
10:22:54 ==> default: ================================================================================
10:22:54 ==> default: Error executing action `modify` on resource 'group[Administrators]'
10:22:54 ==> default: ================================================================================
10:22:54 ==> default: 
10:22:54 ==> default: Chef::Exceptions::Win32APIError
10:22:54 ==> default: -------------------------------
10:22:54 ==> default: No mapping between account names and security IDs was done.
10:22:54 ==> default: ---- Begin Win32 API output ----
10:22:54 ==> default: System Error Code: 1332
10:22:54 ==> default: System Error Message: No mapping between account names and security IDs was done.
10:22:54 ==> default: ---- End Win32 API output ----
10:22:54 ==> default: 
10:22:54 ==> default: Resource Declaration:
10:22:54 ==> default: ---------------------
10:22:54 ==> default: # In C:\vagrant-berkshelf\911e87c11c773d3705a1992318107a59\cookbooks\windows_screenresolution\recipes\default.rb
10:22:54 ==> default: 
10:22:54 ==> default:   1: if platform?('windows')
10:22:54 ==> default:   2:   if node['windows_screenresolution']['rdp_domain'].nil?
10:22:54 ==> default: 
10:22:54 ==> default: Compiled Resource:
10:22:54 ==> default: ------------------
10:22:54 ==> default: # Declared in C:\vagrant-berkshelf\911e87c11c773d3705a1992318107a59\cookbooks\windows_screenresolution\recipes\default.rb:19:in `block in from_file'
10:22:54 ==> default: 
10:22:54 ==> default: group("Administrators") do
10:22:54 ==> default:   action [:modify]
10:22:54 ==> default:   retries 0
10:22:54 ==> default:   retry_delay 2
10:22:54 ==> default:   default_guard_interpreter :default
10:22:54 ==> default:   group_name "Administrators"
10:22:54 ==> default:   members ["rdp_local"]
10:22:54 ==> default:   append true
10:22:54 ==> default:   declared_type :group
10:22:54 ==> default:   cookbook_name :windows_screenresolution
10:22:54 ==> default:   recipe_name "default"
10:22:54 ==> default: end
10:22:54 ==> default: [2016-02-04T10:22:52+00:00] INFO: Running queued delayed notifications before re-raising exception
10:22:54 ==> default: [2016-02-04T10:22:52+00:00] DEBUG: Re-raising exception: Chef::Exceptions::Win32APIError - group[Administrators] (windows_screenresolution::default line 19) had an error: Chef::Exceptions::Win32APIError: No mapping between account names and security IDs was done.
10:22:54 ==> default: ---- Begin Win32 API output ----
10:22:54 ==> default: System Error Code: 1332
10:22:54 ==> default: System Error Message: No mapping between account names and security IDs was done.
10:22:54 ==> default: ---- End Win32 API output ----
10:22:54 ==> default: 
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/win32/error.rb:69:in `raise!'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/win32/security.rb:312:in `lookup_account_name'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/provider/group/windows.rb:93:in `local_group_name_to_sid'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/provider/group/windows.rb:83:in `has_current_group_member?'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/provider/group.rb:96:in `block in compare_group'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/provider/group.rb:95:in `each'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/provider/group.rb:95:in `compare_group'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/provider/group.rb:161:in `action_modify'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/provider.rb:130:in `run_action'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/resource.rb:562:in `run_action'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/runner.rb:49:in `run_action'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/runner.rb:81:in `block (2 levels) in converge'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/runner.rb:81:in `each'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/runner.rb:81:in `block in converge'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/resource_list.rb:83:in `block in execute_each_resource'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:116:in `call'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:116:in `call_iterator_block'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:85:in `step'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:104:in `iterate'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:55:in `each_with_index'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/resource_list.rb:81:in `execute_each_resource'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/runner.rb:80:in `converge'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/client.rb:339:in `block in converge'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/client.rb:334:in `catch'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/client.rb:334:in `converge'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/client.rb:353:in `converge_and_save'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/client.rb:457:in `run'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/application.rb:253:in `run_with_graceful_exit_option'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/application.rb:230:in `block in run_chef_client'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/local_mode.rb:39:in `with_server_connectivity'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/application.rb:213:in `run_chef_client'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/application/solo.rb:280:in `block in interval_run_chef_client'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/application/solo.rb:269:in `loop'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/application/solo.rb:269:in `interval_run_chef_client'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/application/solo.rb:247:in `run_application'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/lib/chef/application.rb:60:in `run'
10:22:54 ==> default:   C:/opscode/chef/embedded/apps/chef/bin/chef-solo:25:in `<top (required)>'
10:22:54 ==> default:   C:/opscode/chef/bin/chef-solo:63:in `load'
10:22:54 ==> default:   C:/opscode/chef/bin/chef-solo:63:in `<main>'
10:22:54 ==> default: [2016-02-04T10:22:52+00:00] ERROR: Running exception handlers
10:22:54 ==> default: [2016-02-04T10:22:52+00:00] ERROR: Exception handlers complete
10:22:54 ==> default: [2016-02-04T10:22:52+00:00] FATAL: Stacktrace dumped to C:/var/chef/cache/chef-stacktrace.out
10:22:54 ==> default: [2016-02-04T10:22:52+00:00] DEBUG: Chef::Exceptions::Win32APIError: group[Administrators] (windows_screenresolution::default line 19) had an error: Chef::Exceptions::Win32APIError: No mapping between account names and security IDs was done.
10:22:54 ==> default: ---- Begin Win32 API output ----
10:22:54 ==> default: System Error Code: 1332
10:22:54 ==> default: System Error Message: No mapping between account names and security IDs was done.
10:22:54 ==> default: ---- End Win32 API output ----
10:22:54 ==> default: 
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/win32/error.rb:69:in `raise!'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/win32/security.rb:312:in `lookup_account_name'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/provider/group/windows.rb:93:in `local_group_name_to_sid'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/provider/group/windows.rb:83:in `has_current_group_member?'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/provider/group.rb:96:in `block in compare_group'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/provider/group.rb:95:in `each'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/provider/group.rb:95:in `compare_group'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/provider/group.rb:161:in `action_modify'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/provider.rb:130:in `run_action'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/resource.rb:562:in `run_action'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/runner.rb:49:in `run_action'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/runner.rb:81:in `block (2 levels) in converge'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/runner.rb:81:in `each'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/runner.rb:81:in `block in converge'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/resource_list.rb:83:in `block in execute_each_resource'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:116:in `call'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:116:in `call_iterator_block'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:85:in `step'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:104:in `iterate'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:55:in `each_with_index'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/resource_list.rb:81:in `execute_each_resource'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/runner.rb:80:in `converge'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/client.rb:339:in `block in converge'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/client.rb:334:in `catch'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/client.rb:334:in `converge'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/client.rb:353:in `converge_and_save'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/client.rb:457:in `run'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/application.rb:253:in `run_with_graceful_exit_option'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/application.rb:230:in `block in run_chef_client'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/local_mode.rb:39:in `with_server_connectivity'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/application.rb:213:in `run_chef_client'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/application/solo.rb:280:in `block in interval_run_chef_client'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/application/solo.rb:269:in `loop'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/application/solo.rb:269:in `interval_run_chef_client'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/application/solo.rb:247:in `run_application'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/lib/chef/application.rb:60:in `run'
10:22:54 ==> default: C:/opscode/chef/embedded/apps/chef/bin/chef-solo:25:in `<top (required)>'
10:22:54 ==> default: C:/opscode/chef/bin/chef-solo:63:in `load'
10:22:54 ==> default: C:/opscode/chef/bin/chef-solo:63:in `<main>'
10:22:54 ==> default: [2016-02-04T10:22:52+00:00] FATAL: Chef::Exceptions::Win32APIError: group[Administrators] (windows_screenresolution::default line 19) had an error: Chef::Exceptions::Win32APIError: No mapping between account names and security IDs was done.
10:22:54 ==> default: 
10:22:54 ==> default: ---- Begin Win32 API output ----
10:22:54 ==> default: System Error Code: 1332
10:22:54 ==> default: System Error Message: No mapping between account names and security IDs was done.
10:22:54 ==> default: ---- End Win32 API output ----
10:22:55 Chef never successfully completed! Any errors should be visible in the
10:22:55 output above. Please fix your recipes so that they properly complete.
10:22:55 **** ERROR ON VAGRANT PROVISION ****
```

The contents of C:/var/chef/cache/chef-stacktrace.out is

```
Generated at 2016-02-04 10:22:52 +0000
Chef::Exceptions::Win32APIError: group[Administrators] (windows_screenresolution::default line 19) had an error: Chef::Exceptions::Win32APIError: No mapping between account names and security IDs was done.
---- Begin Win32 API output ----
System Error Code: 1332
System Error Message: No mapping between account names and security IDs was done.
---- End Win32 API output ----

C:/opscode/chef/embedded/apps/chef/lib/chef/win32/error.rb:69:in `raise!'
C:/opscode/chef/embedded/apps/chef/lib/chef/win32/security.rb:312:in `lookup_account_name'
C:/opscode/chef/embedded/apps/chef/lib/chef/provider/group/windows.rb:93:in `local_group_name_to_sid'
C:/opscode/chef/embedded/apps/chef/lib/chef/provider/group/windows.rb:83:in `has_current_group_member?'
C:/opscode/chef/embedded/apps/chef/lib/chef/provider/group.rb:96:in `block in compare_group'
C:/opscode/chef/embedded/apps/chef/lib/chef/provider/group.rb:95:in `each'
C:/opscode/chef/embedded/apps/chef/lib/chef/provider/group.rb:95:in `compare_group'
C:/opscode/chef/embedded/apps/chef/lib/chef/provider/group.rb:161:in `action_modify'
C:/opscode/chef/embedded/apps/chef/lib/chef/provider.rb:130:in `run_action'
C:/opscode/chef/embedded/apps/chef/lib/chef/resource.rb:562:in `run_action'
C:/opscode/chef/embedded/apps/chef/lib/chef/runner.rb:49:in `run_action'
C:/opscode/chef/embedded/apps/chef/lib/chef/runner.rb:81:in `block (2 levels) in converge'
C:/opscode/chef/embedded/apps/chef/lib/chef/runner.rb:81:in `each'
C:/opscode/chef/embedded/apps/chef/lib/chef/runner.rb:81:in `block in converge'
C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/resource_list.rb:83:in `block in execute_each_resource'
C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:116:in `call'
C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:116:in `call_iterator_block'
C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:85:in `step'
C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:104:in `iterate'
C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:55:in `each_with_index'
C:/opscode/chef/embedded/apps/chef/lib/chef/resource_collection/resource_list.rb:81:in `execute_each_resource'
C:/opscode/chef/embedded/apps/chef/lib/chef/runner.rb:80:in `converge'
C:/opscode/chef/embedded/apps/chef/lib/chef/client.rb:339:in `block in converge'
C:/opscode/chef/embedded/apps/chef/lib/chef/client.rb:334:in `catch'
C:/opscode/chef/embedded/apps/chef/lib/chef/client.rb:334:in `converge'
C:/opscode/chef/embedded/apps/chef/lib/chef/client.rb:353:in `converge_and_save'
C:/opscode/chef/embedded/apps/chef/lib/chef/client.rb:457:in `run'
C:/opscode/chef/embedded/apps/chef/lib/chef/application.rb:253:in `run_with_graceful_exit_option'
C:/opscode/chef/embedded/apps/chef/lib/chef/application.rb:230:in `block in run_chef_client'
C:/opscode/chef/embedded/apps/chef/lib/chef/local_mode.rb:39:in `with_server_connectivity'
C:/opscode/chef/embedded/apps/chef/lib/chef/application.rb:213:in `run_chef_client'
C:/opscode/chef/embedded/apps/chef/lib/chef/application/solo.rb:280:in `block in interval_run_chef_client'
C:/opscode/chef/embedded/apps/chef/lib/chef/application/solo.rb:269:in `loop'
C:/opscode/chef/embedded/apps/chef/lib/chef/application/solo.rb:269:in `interval_run_chef_client'
C:/opscode/chef/embedded/apps/chef/lib/chef/application/solo.rb:247:in `run_application'
C:/opscode/chef/embedded/apps/chef/lib/chef/application.rb:60:in `run'
C:/opscode/chef/embedded/apps/chef/bin/chef-solo:25:in `<top (required)>'
C:/opscode/chef/bin/chef-solo:63:in `load'
C:/opscode/chef/bin/chef-solo:63:in `<main>'
```

**Comment by dhoer** [comment]
When you usually get this error message you have a User ID that is mapped to a security ID (SID) that no longer exists on your computer and is reporting the error.

Each user, group or object has a Security Id or identifier. Example: You may have a user name that was in a application and that user name is no longer valid or has been renamed but the application is still referencing that name. So it reports an error. You may have to go back and change back to the original name or remove the name from the problem application. 

From: http://answers.microsoft.com/en-us/windows/forum/windows_7-windows_programs/no-mapping-between-security-id-error-in-windows-7/9cd51407-b965-4649-aa55-2ec582aa5e60

**Comment by chris-orchard** [comment]
Thanks for the advice although I'm somewhat confused why I'm now getting this error instead. Also your 'click here' article link isn't a link! :)

As well as turning on debug I updated to chef 12.6.0 to see if this fixed the error as you thought which was when I started getting the different error above.

I've now gone back to chef 12.3.0 and taken out the debug flag so am back how I started and expected to get the original 'invalid backref' error but still get the one above.

The VM is destroyed and recreated every chef run so am confused why it is a different error when I everything else should be the same as it was.

I'll update if I get anywhere.

**Comment by dhoer** [comment]
That link is from cut and paste.  I removed it.  I'm not sure why you are getting this issue. I haven't run into this problem, but I use Chef 11.18. 

See if you get the same issue with this Windows box:

``` ruby
platforms:
  - name: windows-2012r2
    driver:
      box: dhoer/windows-2012r2
      network:
      - ["forwarded_port", {guest: 4444, host: 4446}]
    attributes:
```

**Comment by chris-orchard** [comment]
I haven't tried your box yet, I'll do that next.

However I have narrowed down that the issue seems to be when it attempts to add the newly created rdp_local user to Administrators group:

```
Error executing action `modify` on resource 'group[Administrators]'
```

I manually made the rdp_local user an Administrator through control panel, rebooted, then ran chef again and it reported this step as up to date, and proceeded to the end of the cookbook without further errors.

I then manually changed rdp_local back to a standard user, ran chef again, this time it successfully managed to add rdp_local back to administrators group.

I'm just re-provisioning the VM so I can check if it was the reboot, or manually making it an admin that worked round the issue.

**Comment by chris-orchard** [comment]
It seems to be the reboot that got it working. 

I just re-provisioned the vm, got the 1332 err, rebooted it, logged on and ran chef again and got:

```
Recipe: windows_screenresolution::default
  * user[rdp_local] action create (up to date)
  * group[Administrators] action modify
    - modify group Administrators
    - add missing member(s): rdp_local
  * group[Remote Desktop Users] action modify
    - modify group Remote Desktop Users
    - add missing member(s): rdp_local
  * windows_home[rdp_local] action create
    - Generate user's home directory C:\Users\rdp_local

Is there a reason it would need a reboot after creating rdp_local user before it could add it to Administrators group? If so how could I get this working in one pass?
```

**Comment by dhoer** [comment]
It shouldn't need to.  I don't have this issue on Chef 11 where we build 70+ Windows machines daily. 

Can you try cloning git@github.com:dhoer/chef-windows_home.git and run `kitchen test` against your Windows box to see if it completes successfully?

This is what creates the user's directory structure.  There is a bug with it hanging if you are on a unplugged laptop, but that is the issue reported.

**Comment by chris-orchard** [comment]
I've tried chef 11.18.12 and setting `default['windows_screenresolution']['rdp_username']` and `default['windows_screenresolution']['rdp_password']` to an existing user who is already an administrator instead of letting it create the default 'rdp_local' user.

This still has the same issue with `group[Administrators] action modify` even though this user is already in the administrator group so should be up to date. Again after rebooting the VM and running again the `group[Administrators] action modify` works with up-to-date as expected.

Just trying kitchen on the above now.

**Comment by chris-orchard** [comment]
I've tried my cookbook on your `dhoer/windows-2012r2` box as suggested in [your earlier comment](https://github.com/dhoer/chef-windows_screenresolution/issues/4#issuecomment-179931225) and it runs through without error, including the windows_screenresolution cookbook.

I'm not sure how to try [your other suggestion](https://github.com/dhoer/chef-windows_screenresolution/issues/4#issuecomment-179998084) of running test kitchen with our Windows box as it's a VSphere box and I've not managed to get kitchen to work against VSphere.

However I've been able to work round the issue by doing a `vagrant reload --provision` from our Jenkins job that creates the selenium VMs, the vagrant reload doing the restart that seems to be required on our Windows box before the set administrator action of the cookbook will work.

As I've worked round this, and the original "invalid backref number" issue I actually raised went away, feel free to close this. Sorry that I couldn't provide a reproducible or confirm for sue if it was our Windows box rather than the cookbook or a chef issue.
